src/Components/MultiInput/Component.purs

```diff
@@ -494,6 +496,40 @@ removeItem index = do
   when (Data.Array.null new.items) do
     appendNewItem
 
+selectItem ::
+  forall m.
+  MonadAff m =>
+  String ->
+  ComponentM m Unit
+selectItem text = do
+  old <- Halogen.get
+  void $ Control.Monad.Maybe.Trans.runMaybeT do
+    index <-
+      Control.Monad.Maybe.Trans.MaybeT <<< pure
+        $ Data.Array.findIndex isEditable old.items
+    width <-
+      Control.Monad.Maybe.Trans.MaybeT
+         $ measureTextWidth text
+    void <<< Control.Monad.Maybe.Trans.lift
+        $ updateItem index
+          ( \item -> case item of
+              Display _ -> item
+              Edit status -> Edit status { inputBox = { width, text } }
+              New status -> New status { inputBox = { width, text } }
+          )
+    Control.Monad.Maybe.Trans.lift do
+      commitEditing index
+      when (isLastIndex index old.items) do
+        new <- Halogen.get
+        focusItem (getLastIndex new.items)
+
```

Comment:
🎨  I feel like this could also be simplified to avoid transformers in a similar way to the previous suggestion.